helper/identity/templating_test.go

```diff
@@ -189,10 +368,97 @@ func TestPopulate_Basic(t *testing.T) {
 			}
 		}
 		if out != test.output {
-			t.Fatalf("%s: bad output: %s", test.name, out)
+			t.Fatalf("%s: bad output: %s, expected: %s", test.name, out, test.output)
 		}
 		if err == nil && !subst && out != test.input {
 			t.Fatalf("%s: bad subst flag", test.name)
 		}
 	}
 }
+
+func TestPopulate_FullObject(t *testing.T) {
+	var testEntity = &Entity{
+		ID:   "abc-123",
+		Name: "Entity Name",
+		Metadata: map[string]string{
+			"color":         "green",
+			"size":          "small",
+			"non-printable": "\"\n\t",
+		},
+		Aliases: []*Alias{
+			{
+				MountAccessor: "aws_123",
+				Metadata: map[string]string{
+					"service": "ec2",
+					"region":  "west",
+				},
+			},
+		},
+	}
+
+	var testGroups = []*Group{
+		{ID: "a08b0c02", Name: "g1"},
+		{ID: "239bef91", Name: "g2"},
+	}
+
+	template := `
+			{
+			    "id": {{identity.entity.id}},
+			    "name": {{identity.entity.name}},
+			    "all metadata": {{identity.entity.metadata}},
+			    "one metadata key": {{identity.entity.metadata.color}},
+			    "one metadata key not found": {{identity.entity.metadata.asldfk}},
+			    "alias metadata": {{identity.entity.aliases.aws_123.metadata}},
+			    "alias not found metadata": {{identity.entity.aliases.blahblah.metadata}},
+			    "one alias metadata key": {{identity.entity.aliases.aws_123.metadata.service}},
+			    "one not found alias metadata key": {{identity.entity.aliases.blahblah.metadata.service}},
+			    "group names": {{identity.entity.group_names}},
+			    "group ids": {{identity.entity.group_ids}},
+			    "time now": {{time.now}},
+			    "time plus": {{time.now.plus.10s}},
+			    "time minus": {{time.now.minus.1h}},
+			    "repeated and": {"nested element": {{identity.entity.name}}}
+			}`
+
+	expected := `
+			{
+			    "id": "abc-123",
+			    "name": "Entity Name",
+			    "all metadata": {"color":"green","non-printable":"\"\n\t","size":"small"},
+			    "one metadata key": "green",
+			    "one metadata key not found": "",
+			    "alias metadata": {"region":"west","service":"ec2"},
+			    "alias not found metadata": {},
+			    "one alias metadata key": "ec2",
+			    "one not found alias metadata key": "",
+			    "group names": ["g1","g2"],
+			    "group ids": ["a08b0c02","239bef91"],
+			    "time now": <now>,
+			    "time plus": <plus>,
+			    "time minus": <minus>,
+			    "repeated and": {"nested element": "Entity Name"}
+			}`
+
+	now := time.Now().Unix()
+	minus := now - (60 * 60)
+	plus := now + 10
+
+	expected = strings.ReplaceAll(expected, "<now>", strconv.FormatInt(now, 10))
```

Comment:
This is racy.  Can you add an optional PopulateStringInput field used for testing that specifies what "now" is?

Reply:
Yeah. I think a Now member has use even beyond testing.

Reply:
Added